src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Extensions/OperationExtensions.cs

```diff
@@ -210,7 +210,16 @@ where void Foo(in T v)
             // Accessing an indexer/property off of a value type will read/write the value type depending on how the
             // indexer/property itself is used. We add ValueUsageInfo.Read to the result since the value on which we are
             // accessing a property is always read.
-            return ValueUsageInfo.Read | GetValueUsageInfo(operation.Parent, containingSymbol);
+            // We consider a potential write only if the type is not a ref-like type.
```

Comment:
I think it would be good to explain why being a ref-like type matters. Like why a write through it is still not considered a write. I'm genuinely not clear on why that's the right way to think about things (even if it helps this bug).

Reply:
I expanded the doc comment on how I'm thinking about it. I can't say I'm 100% confident though.

Reply:
The explanation definitely helped!